File: 1752-check-if-array-is-sorted-and-rotated/1752-check-if-array-is-sorted-and-rotated.py
class Solution:
    def check(self, nums: List[int]) -> bool:
        x = 0
        
        for i in range(len(nums)-1):
            if nums[i] > nums[i+1]:
                x = i + 1
                break
        sorted_nums = sorted(nums)
        
        for j in range(len(nums)):
            if sorted_nums[j] != nums[(j+x)% len(nums)]:
                return False
        return True

File: 1752-check-if-array-is-sorted-and-rotated/1752-check-if-array-is-sorted-and-rotated.java
class Solution {
    public boolean check(int[] nums) {
        int size = nums.length;
        int count = 0;

        for (int i = 0; i < size - 1; i++){
            if (nums[i] > nums[i+1]){
                count++;
            } 
        }

        return (count == 0) || (count == 1 && nums[0] >= nums[size -1]);
    }
}